spacy/default_config.cfg

```diff
@@ -2,6 +2,8 @@
 lang = null
 pipeline = []
 load_vocab_data = true
+before_init = null
```

Comment:
The big question is whether we also want to run the hooks at runtime, or is this a training-only step?

I think they do also need to run at runtime, right?

Reply:
I think they should also run a runtime, otherwise it would be a bit confusing and means people need separate workflows. If you're customising the tokenizer with a registered function, you should also be able to provide your before/after callbacks like this.

(Also, I still have this vision of somehow providing a workflow for users to automatically include their Python code provided via `--code` into their packaged model. That would solve the whole "train with custom code and ship your model" question...)

Reply:
I think that should be feasible, but as an argument of `spacy package` not `spacy train`, right? In package we can either just dump it into the `__init__`, or import it within the init.

Reply:
Yeah, exactly, that was the idea. `spacy train` doesn't package, so including the code anywhere wouldn't make sense here. But you can pass it in when you package and we can import it in the `__init__.py` (if it's all one module and self-contained, which makes sense here anyways).